wallet/src/main/java/com/android/identity_credential/wallet/presentation/TransferHelper.kt

```diff
@@ -142,17 +157,20 @@ class TransferHelper(
         requestedDocType: String,
         credentialId: String,
         credentialRequest: CredentialRequest,
-        onFinishedProcessing: (ByteArray) -> Unit
-    ) = withContext(Dispatchers.IO) {
+        keyUnlockData: KeyUnlockData?,
+        onFinishedProcessing: (ByteArray) -> Unit,
+        onResultReady: (result: AddDocumentToResponseResult) -> Unit,
+    ) {
```

Comment:
Hmm, first of all we, when we modify code we also need to modify the existing docs otherwise it gets very confusing. From a 50,000 point of view one point here is that we don't know a'priori whether authentication is needed to generate the response... so at least "This is called once the user accepted various prompts (consent, biometrics, etc..)" needs to be updated in the docs and we also need to add param/returns.

Second, the callback you introduce is called `onResultReady` but... you call this in the event that `KeyLockedException` is thrown...it looks like this stems from lot of copy-paste from appHolder which adds to the confusion especially since it uses words like "Document" which we don't use in this app.

So I think `finishProcessingRequest()` should be modified in the following way

- it should have a way to return that user-auth is needed... for this, it also needs to pass the `AuthenticationKey` instance so the caller can see what kind of auth is needed and so we can fish out the `CryptoObject` and invoke the biometric prompt like we do today. Maybe introduce a callback that takes an AuthenticationKey and KeyLockedException and call it `onAuthenticationKeyLocked()`

- I think you also need to be able to hand back the same authkey since, well, this is in fact the one that was authenticated... instead we seem to always pick a new one below with `credential.findAuthenticationKey()` ... the only reason it works right now is a bug in `AndroidKeystoreSecureArea` that we should fix... anyway, maybe call this parameter `keyToUnlock: AuthenticationKey?` and have it sit next to the `KeyUnlockData?`